tests/dump_tests/match_engine_test.py

```diff
@@ -182,75 +217,60 @@ def test_just_keys_false(self):
         assert not ddiff, ddiff
 
-    def test_file_source(self):
+    def test_file_source(self, match_engine):
         file = os.path.join(dump_test_input, "copp_cfg.json")
         req = MatchRequest(file=file, table="COPP_TRAP", field="trap_ids", value="arp_req")
-        ret = self.match_engine.fetch(req)
+        ret = match_engine.fetch(req)
         assert ret["error"] == ""
         assert len(ret["keys"]) == 1
         assert "COPP_TRAP|arp" in ret["keys"]
 
-    def test_file_source_with_key_ptrn(self):
+    def test_file_source_with_key_ptrn(self, match_engine):
         file = os.path.join(dump_test_input, "copp_cfg.json")
         req = MatchRequest(file=file, table="COPP_GROUP", key_pattern="queue4*", field="red_action", value="drop")
-        ret = self.match_engine.fetch(req)
+        ret = match_engine.fetch(req)
         assert ret["error"] == ""
         assert len(ret["keys"]) == 1
         assert "COPP_GROUP|queue4_group2" in ret["keys"]
 
-    def test_file_source_with_not_only_return_keys(self):
+    def test_file_source_with_not_only_return_keys(self, match_engine):
         file = os.path.join(dump_test_input, "copp_cfg.json")
         req = MatchRequest(file=file, table="COPP_GROUP", key_pattern="queue4*", field="red_action", value="drop", just_keys=False)
-        ret = self.match_engine.fetch(req)
+        ret = match_engine.fetch(req)
         assert ret["error"] == ""
         assert len(ret["keys"]) == 1
         recv_dict = ret["keys"][0]
         exp_dict = {"COPP_GROUP|queue4_group2": {"trap_action": "copy", "trap_priority": "4", "queue": "4", "meter_type": "packets", "mode": "sr_tcm", "cir": "600", "cbs": "600", "red_action": "drop"}}
         ddiff = DeepDiff(exp_dict, recv_dict)
         assert not ddiff, ddiff
 
-    def test_match_entire_list(self):
+    def test_match_entire_list(self, match_engine):
         req = MatchRequest(db="CONFIG_DB", table="PORT", key_pattern="*", field="lanes", value="61,62,63,64", match_entire_list=True, just_keys=True)
-        ret = self.match_engine.fetch(req)
+        ret = match_engine.fetch(req)
         assert ret["error"] == ""
         assert len(ret["keys"]) == 1
         assert "PORT|Ethernet60" in ret["keys"]
 
+@pytest.mark.usefixtures("match_engine")
+class TestNonDefaultNameSpace:
 
-class TestNonDefaultNameSpace(unittest.TestCase):
-
-    @classmethod
-    def setup_class(cls):
-        print("SETUP")
-        os.environ["UTILITIES_UNIT_TESTING"] = "2"
-        os.environ["UTILITIES_UNIT_TESTING_TOPOLOGY"] = "multi_asic"
-        from ..mock_tables import mock_multi_asic
-        reload(mock_multi_asic)
-        from ..mock_tables import dbconnector
-        dbconnector.load_namespace_config()
-
-    def teardown_class(cls):
-        print("TEARDOWN")
-        os.environ["UTILITIES_UNIT_TESTING"] = "0"
-        os.environ["UTILITIES_UNIT_TESTING_TOPOLOGY"] = ""
-
-    def test_namespace_asic0(self):
+    def test_namespace_asic0(self, match_engine):
```

Comment:
How is the config_db data from namespace fetched if  setup is removed to load namespace config?

Reply:
How is match_engine toggling between single namespace and mulitple namespace scenario?

Reply:
I'm pre-populating ConnectionPool based on namespace as the key ahead of time and using it to construct MatchEngine. 

Once the Sonicv2Connector Objects are cached in the ConnectionPool, MatchEngine doesn't re-initialize them but will rather use the existing ones and thus there is no different b/w both the scenarios.

Reply:
so this testing will be checking data in DEFAULT_NAMESPACE and a specific namespace like namepsace asic0, which will be like a multi-asic DUT.
Earlier, the unit-test was testing on DEFAULT_NAMESPACE of single asic DUT and asic0 of multi-asic DUT. correct?

Reply:
Yeah, based on the ns argument provided to MatchRequest. The differentiation is actually made/required in dump_state_test.py (These tests invoke the command using the )

There you could see i've used different fixtures for masic and single asic